lib/Transformations.cpp

```diff
@@ -217,4 +191,35 @@ namespace CIL {
             img.setColorModel(ColorModel::COLOR_GRAY);
         img.setData(new_img_data);
     }
+
+    void resize(ImageInfo& img, uint32_t new_width, uint32_t new_height,
+                ResizeAlgorithm resize_algorithm)
+    {
+        double scale_x = (new_width * 1.00) / img.width();
+        double scale_y = (new_height * 1.00) / img.height();
+
+        ImageMatrix new_image_data(new_width, new_height, img.numComponents(),
+                                   img.sampleDepth());
+
+        for (auto px : new_image_data)
+        {
+            Coordinate source = {px.col() / scale_x, px.row() / scale_y};
+            switch (resize_algorithm)
+            {
+                case ResizeAlgorithm::nearest_neighbour_interpolation:
+                {
+                    auto old_px = img(source.getNearestIntegralY(),
+                                      source.getNearestIntegralX());
+                    if (old_px.isValid())
+                        px.copyComponents(old_px);
+                    break;
+                }
+                case ResizeAlgorithm::bilinear_interpolation:
+                    px = utils::bilinearInterpolation(img, source);
+                    break;
+                case ResizeAlgorithm::bicubic_interpolation: break;
```

Comment:
I will prefer that this isn't blank, just put an `assert("not implemented")`.

Reply:
I have altogether removed the case statement corresponding to `bicubic_interpolation`.